### PR TITLE
Fix flaky periodic tests

### PR DESCRIPTION
**Pull request by dusk125**
Addresses https://github.com/etcd-io/etcd/issues/17391

Removes the need to timeout in real-time by using the `clockwork.FakeClock.BlockUtil` function to wait for the Periodic compactor to get to the Retry "After" step.

### CODE REVIEW

**Comment by dusk125** [comment]
@ahrtr ptal

**Comment by qincwang** [comment]
Hi @dusk125 I am trying the same code in my local and run with `stress` to test flakiness, get this log after goroutine panic, the compaction succeeds and logs look legit though. could you please verify from your end, thank you
image

**Comment by ahrtr** [comment]
Sorry for the late response. @dusk125 could you respond to @qincwang 's comment above?

Please rebase this PR (we merges lots of PR recently), I will take a closer look at this PR sometime next week.

File: client/pkg/testutil/recorder.go [hunk 1]
```diff
@@ -137,3 +137,38 @@ func newLenErr(expected int, actual int) error {
...
+			acts = append(acts, act)
+		default:
+			return acts
+		}
+	}
+}
+
+func (r *recorderSync) Chan() <-chan Action {
+	return r.ch
+}
```
Comment (ahrtr) [inline client/pkg/testutil/recorder.go:166]: The implementation is identical to `recorderStream`.

File: client/pkg/testutil/recorder.go [hunk 2]
```diff
@@ -137,3 +137,38 @@ func newLenErr(expected int, actual int) error {
...
+
+func (r *recorderSync) Wait(n int) ([]Action, error) {
+	acts := make([]Action, 0, n)
+	for i := 0; i < n; i++ {
+		acts = append(acts, <-r.ch)
+	}
+	return acts, nil
+}
```
Comment (ahrtr) [inline client/pkg/testutil/recorder.go:174]: The implementation in https://github.com/etcd-io/etcd/pull/17513 can also resolve this by setting an unlimited timeout.

**Comment by ahrtr** [comment]
Thanks for the fix, but Overall I prefer to https://github.com/etcd-io/etcd/pull/17513 over this PR.

**Comment by dusk125** [comment]
No problem, we can go with #17513 !

**Pull request closed without merging**